lane.py

```diff
@@ -2591,13 +2343,17 @@ def uses_customlists(self):
 
     def update_size(self, _db):
         """Update the stored estimate of the number of Works in this Lane."""
-        query = self.works_from_database(_db).limit(None)
-        query = query.distinct(mw.works_id)
+        library = self.get_library(_db)
 
         # Do the estimate for every known entry point.
         by_entrypoint = dict()
         for entrypoint in EntryPoint.ENTRY_POINTS:
-            qu = entrypoint.apply(query)
+            facets = DatabaseBackedFacets(
+                library, FacetConstants.COLLECTION_FULL,
+                FacetConstants.AVAILABLE_ALL,
+                order=FacetConstants.ORDER_WORK_ID, entrypoint=entrypoint
+            )
+            qu = self.works_from_database(_db, facets)
```

Comment:
Again, I don't think it would be too difficult to use the search index to count the size of a lane under different entry points.